File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/Form/Features/AcceptAndCancelButtonsPage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.Form.Features
{
    public partial class AcceptAndCancelButtonsPage : UserControl
    {
        public AcceptAndCancelButtonsPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles Click event for the button.
        /// Opens the form with assigned accept and cancel buttons.
        /// </summary>
        private void mobjButton_Click(object sender, EventArgs e)
        {
            AcceptAndCancelButtonsForm mobjAcceptAndCancelForm = new AcceptAndCancelButtonsForm();
            mobjAcceptAndCancelForm.Closed += new System.EventHandler(this.mobjAcceptAndCancelForm_Closed);
            mobjAcceptAndCancelForm.ShowDialog();
        }

        /// <summary>
        /// Handles closed event for the form with assigned accept and cancel buttons.
        /// Check the dialog result for the form and 
        /// shows saving or canceling message according to the dialog result.
        /// </summary>
        private void mobjAcceptAndCancelForm_Closed(object sender, EventArgs e)
        {
            AcceptAndCancelButtonsForm mobjAcceptAndCancelButtonsForm = sender as AcceptAndCancelButtonsForm;
            if (mobjAcceptAndCancelButtonsForm.DialogResult == DialogResult.OK)
            {
                MessageBox.Show(mobjAcceptAndCancelButtonsForm.GetSavingUserMessage());
            }
            else
            {
                MessageBox.Show("Cancel editing user data...");
            }

        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/Form/Features/AcceptAndCancelButtonsPage.vb
Namespace CompanionKit.Controls.Form.Features

	Public Class AcceptAndCancelButtonsPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

        ''' <summary>
        ''' Handles Click event for the button.
        ''' Opens the form with assigned accept and cancel buttons.
        ''' </summary>
        Private Sub mobjButton_Click(sender As Object, e As EventArgs) Handles mobjButton.Click
            Dim mobjAcceptAndCancelForm As New AcceptAndCancelButtonsForm()
            AddHandler mobjAcceptAndCancelForm.Closed, AddressOf mobjAcceptAndCancelForm_Closed
            mobjAcceptAndCancelForm.ShowDialog()
        End Sub

        ''' <summary>
        ''' Handles closed event for the form with assigned accept and cancel buttons.
        ''' Check the dialog result for the form and
        ''' shows saving or canceling message according to the dialog result.
        ''' </summary>
        Private Sub mobjAcceptAndCancelForm_Closed(sender As Object, e As EventArgs)
            Dim mobjAcceptAndCancelButtonsForm As AcceptAndCancelButtonsForm = TryCast(sender, AcceptAndCancelButtonsForm)
            If mobjAcceptAndCancelButtonsForm.DialogResult = DialogResult.OK Then
                MessageBox.Show(mobjAcceptAndCancelButtonsForm.GetSavingUserMessage())
            Else
                MessageBox.Show("Cancel editing user data...")
            End If

        End Sub

	End Class

End Namespace